app/src/main/java/com/loc/newsapp/presentation/mainActivity/MainViewModel.kt

```diff
@@ -0,0 +1,41 @@
+package com.loc.newsapp.presentation.mainActivity
+
+import androidx.compose.runtime.State
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.loc.newsapp.domain.manager.manager.LocalUserManager
+import com.loc.newsapp.domain.manager.usecases.AppEntryUseCases
+import com.loc.newsapp.presentation.navgraph.Route
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import javax.inject.Inject
+
+@HiltViewModel
+class MainViewModel @Inject constructor(
+    private val appEntryUseCases: AppEntryUseCases
+): ViewModel() {
+
+    private val _splashCondition = mutableStateOf(true)
+    val splashCondition: State<Boolean> = _splashCondition
```

Comment:
Please check best practices if using the State in ViewModel is okay ? or should be replaced with liveData.